9-type-aliases-redux/latex/kind-judgements.tex

```diff
@@ -0,0 +1,414 @@
+\newcommand{\dtdefine}[2]{\textt{type bindings } #1 \textt{ in } #2}
+
+\begin{document}
+
+\begin{figure}[t]
+	$\arraycolsep=4pt\begin{array}{rllllll}
+			\mathsf{BinOp}               & \oplus & ::= &
+			\textt{Product} ~\vert~ \textt{Sum} ~\vert~ \textt{Arrow}                                          \\
+			\mathsf{Kind}                & \kappa & ::= &
+			\Ty ~\vert~ \KHole ~\vert~ \KSing{\tau}                                                            \\
+			\mathsf{Constant Types}      & c      & ::= &
+			\textt{Int} ~\vert~ \textt{Float} ~\vert~ \textt{Bool}                                             \\
+			\mathsf{User HTyp}           & \htau  & ::= &
+			c ~\vert~ \htau_1 \oplus \htau_2 ~\vert~ \hlist{\htau} ~\vert~ \hthole{u} ~\vert~ \hhole{\htau}{u} \\
+			\mathsf{Internal HTyp}       & \dtau  & ::= &
+			c ~\vert~ \dtau_1 \oplus \dtau_2 ~\vert~ \hlist{\dtau} ~\vert~ \hthole{u} ~\vert~ \hhole{\dtau}{u} \\
+			\mathsf{Type Vars}           & t      &     &                                                      \\
+			\mathsf{User Type Pattern}   & \hrho  & ::= &
+			t ~\vert~ \hthole{u} ~\vert~ \hhole{t}{u}                                                          \\
+			\mathsf{User Expression}     & \hexp  & ::= &
+			\htdefine{\hrho}{\htau}{\hexp} ~\vert~ elided                                                      \\
+			\mathsf{Internal Expression} & \dtau  & ::= &
+			\dtdefine{\hPhi}{\dexp} ~\vert~ elided                                                             \\
+		\end{array}$
+\end{figure}
+
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kconsubkind{\Delta;\hPhi}{\hkappa_1}{\hkappa_2}}
+	{$\hkappa_1$ is a consistent subkind of $\hkappa_2$}
+	\begin{mathpar}
+		\inferrule[KCHoleL]{ }{
+			\kconsubkind{\Delta;\hPhi}{\KHole}{\hkappa}
+		}
+		\and
+		\inferrule[KCHoleR]{ }{
+			\kconsubkind{\Delta;\hPhi}{\hkappa}{\KHole}
+		}
+		\and
+		\inferrule[KCRespectEquiv]{
+			\kequiv{\Delta;\hPhi}{\hkappa_1}{\hkappa_2} \\
+		} {
+			\kconsubkind{\Delta;\hPhi}{\hkappa_1}{\hkappa_2}
+		}
+		\and
+		\inferrule[KCSubsumption]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\Ty}
+		} {
+			\kconsubkind{\Delta;\hPhi}{\KSing{\dtau}}{\Ty}
+		}
+	\end{mathpar}
+\end{minipage}
+\\
+\\
+
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\tyvarValid{t}}
+	{$t$ is a valid type variable} \; \\
+	$t$ is valid if it is not a builtin-type or keyword, begins with an alpha char or underscore, and only contains alphanumeric characters, underscores, and primes.
+\end{minipage}
+\\
+\\
+
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kformation{\Delta;\hPhi}{\hkappa}}
+	{$\hkappa$ forms a kind} \;\\
+	\begin{mathpar}
+		\inferrule[KFSing]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\Ty}
+		} {
+			\kformation{\Delta;\hPhi}{\KSing{\dtau}}
+		}
+	\end{mathpar}
+\end{minipage}
+\\
+\\
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kequiv{\Delta;\hPhi}{\hkappa_1}{\hkappa_2}}
+	{$\hkappa_1$ is equivalent to $\hkappa_2$} \;\\
```

Comment:
need rules for other type forms

Reply:
I'm confused what is you mean by other type forms -- I'm not clear what's missing here